### PR TITLE
Add delete team endpoint

### PR DESCRIPTION
**Pull request by chotai**
Add delete team endpoint accessible by admins
Remove AAD calls from within mongo transactions as this can not be undone
Replace update calls with findOneAndUpdate for efficiency

### CODE REVIEW

File: src/api/teams/controllers/add-user-to-team.js [hunk 1]
```diff
@@ -23,19 +22,19 @@ const addUserToTeamController = {
...
     const dbUser = await getUser(request.db, userId)
 
-    if (isNull(dbTeam) || isNull(dbUser)) {
-      throw Boom.notFound('User or Team not found')
+    if (!dbTeam) {
+      throw Boom.notFound('Team not found')
+    } else if (!dbUser) {
+      throw Boom.notFound('User not found')
     } else if (teamHasUser(dbTeam, dbUser)) {
       throw Boom.conflict('User already a member of the team')
```
Comment (flurdy) [inline src/api/teams/controllers/add-user-to-team.js:30]: I know this wasn't changed in this PR. But why boom conflict  if already there. idempotency and all that. :shrug:

**Review by feedmypixel — COMMENTED** [review]
Feels like it would be good to wrap some tests of value around this.

Especially as its quite an important thing to make sure its correct and that it doesn;t change with new additions

File: src/api/teams/controllers/add-user-to-team.js (hunk 1 above)
Comment (chotai) [reply src/api/teams/controllers/add-user-to-team.js]: Agreed - have updated

**Review by feedmypixel — APPROVED** [review]
LGTM 👍 🚀 🎸

File: src/api/users/controllers/delete-user.js [hunk 2]
```diff
@@ -20,7 +21,20 @@ const deleteUserController = {
   },
   handler: async (request, h) => {
     try {
-      await deleteUser(request, request.params?.userId)
+      const userId = request.params?.userId
+      const user = await deleteUser(request, userId)
+      if (user.teams?.length) {
```
Comment (flurdy) [inline src/api/users/controllers/delete-user.js:26]: does `user` need a `user?` in case line 24 is undefined?

Comment (flurdy) [reply src/api/users/controllers/delete-user.js:26]: or does `user.teams?` handle both?

Comment (flurdy) [reply src/api/users/controllers/delete-user.js:26]: Ah I guess line 25 Booms an exception if not found

**Review by flurdy — APPROVED** [review]

**Pull request merged**